src/StackExchange.Redis/Enums/RedisCommand.cs

```diff
@@ -1,236 +1,465 @@
+            case RedisCommand.HSETNX:
+            case RedisCommand.INCR:
+            case RedisCommand.INCRBY:
+            case RedisCommand.INCRBYFLOAT:
+            case RedisCommand.LINSERT:
+            case RedisCommand.LMOVE:
+            case RedisCommand.LPOP:
+            case RedisCommand.LPUSH:
+            case RedisCommand.LPUSHX:
+            case RedisCommand.LREM:
+            case RedisCommand.LSET:
+            case RedisCommand.LTRIM:
+            case RedisCommand.MIGRATE:
+            case RedisCommand.MOVE:
+            case RedisCommand.MSET:
+            case RedisCommand.MSETNX:
+            case RedisCommand.PERSIST:
+            case RedisCommand.PEXPIRE:
+            case RedisCommand.PEXPIREAT:
+            case RedisCommand.PEXPIRETIME:
+            case RedisCommand.PFADD:
+            case RedisCommand.PFMERGE:
+            case RedisCommand.PSETEX:
+            case RedisCommand.RENAME:
+            case RedisCommand.RENAMENX:
+            case RedisCommand.RESTORE:
+            case RedisCommand.RPOP:
+            case RedisCommand.RPOPLPUSH:
+            case RedisCommand.RPUSH:
+            case RedisCommand.RPUSHX:
+            case RedisCommand.SADD:
+            case RedisCommand.SDIFFSTORE:
+            case RedisCommand.SET:
+            case RedisCommand.SETBIT:
+            case RedisCommand.SETEX:
+            case RedisCommand.SETNX:
+            case RedisCommand.SETRANGE:
+            case RedisCommand.SINTERSTORE:
+            case RedisCommand.SMOVE:
+            case RedisCommand.SPOP:
+            case RedisCommand.SREM:
+            case RedisCommand.SUNIONSTORE:
+            case RedisCommand.SWAPDB:
+            case RedisCommand.TOUCH:
+            case RedisCommand.UNLINK:
+            case RedisCommand.ZADD:
+            case RedisCommand.ZDIFFSTORE:
+            case RedisCommand.ZINTERSTORE:
+            case RedisCommand.ZINCRBY:
+            case RedisCommand.ZPOPMAX:
+            case RedisCommand.ZPOPMIN:
+            case RedisCommand.ZRANGESTORE:
+            case RedisCommand.ZREM:
+            case RedisCommand.ZREMRANGEBYLEX:
+            case RedisCommand.ZREMRANGEBYRANK:
+            case RedisCommand.ZREMRANGEBYSCORE:
+            case RedisCommand.ZUNIONSTORE:
+                return true;
+            // Commands that can be issued anywhere
+            case RedisCommand.NONE:
+            case RedisCommand.ASKING:
+            case RedisCommand.AUTH:
+            case RedisCommand.BGREWRITEAOF:
+            case RedisCommand.BGSAVE:
+            case RedisCommand.BITCOUNT:
+            case RedisCommand.BITPOS:
+            case RedisCommand.CLIENT:
+            case RedisCommand.CLUSTER:
+            case RedisCommand.CONFIG:
+            case RedisCommand.DBSIZE:
+            case RedisCommand.DEBUG:
+            case RedisCommand.DISCARD:
+            case RedisCommand.DUMP:
+            case RedisCommand.ECHO:
+            case RedisCommand.EVAL:
```

Comment:
Eval and EvalSha can be write commands I'm pretty sure they would be rejected by a replica. That's the impetus behind the RO variants

Reply:
For this purpose, we only want to enforce if it _will_ fail, so want to leave the eval bits which we can't tell. For example if we changed those, it may change someone's production load from being spread over 20 replicas to slamming a primary.

Reply:
yep, you're right, reading [this](https://redis.io/docs/manual/replication/#read-only-replica) I thought that eval/evalsha would be considered write commands in and of themselves. [Evidently not](https://github.com/redis/redis/blob/unstable/src/commands.c#L3290) - you'd see `CMD_WRITE` in the flags.